source/postprocess/particles.cc

```diff
@@ -31,21 +32,119 @@ namespace aspect
 {
   namespace Postprocess
   {
+    namespace internal
+    {
+      template<int dim>
+      ParticleOutput<dim>::~ParticleOutput()
+      {}
+
+      template<int dim>
+      void
+      ParticleOutput<dim>::build_patches(const std::multimap<aspect::Particle::types::LevelInd, aspect::Particle::Particle<dim> > &particles,
+                                         const aspect::Particle::Property::ParticlePropertyInformation &property_information)
+      {
+        // First store the names of the data fields
+        dataset_names.reserve(property_information.n_components()+1);
+        dataset_names.push_back("id");
+
+        for (unsigned int field_index = 0; field_index < property_information.n_fields(); ++field_index)
+          {
+            const unsigned n_components = property_information.get_components_by_field_index(field_index);
+            const std::string field_name = property_information.get_field_name_by_index(field_index);
+            // If it is a 1D element, or a vector, print just the name, otherwise use []
+            if ((n_components == 1) || (n_components == dim))
+              for (unsigned int component_index=0; component_index<n_components; ++component_index)
+                dataset_names.push_back(field_name);
+            else
+              for (unsigned int component_index=0; component_index<n_components; ++component_index)
+                dataset_names.push_back(field_name + "[" + Utilities::to_string(component_index) + "]");
```

Comment:
I don't think `[` and `]` are valid names for data set names. You may have to just append the index with an underscore.